app/src/lib/stores/pull-request-store.ts

```diff
@@ -38,11 +81,43 @@ export class PullRequestStore {
     }
 
     const pullRequests = await this.db.pullRequests
-      .where('repo_id')
+      .where('base.repoId')
       .equals(gitHubRepositoryID)
+      .reverse()
       .sortBy('number')
 
-    return pullRequests
+    const builtPullRequests = new Array<PullRequest>()
+    for (const pr of pullRequests) {
+      const headId = pr.head.repoId
+      let head: GitHubRepository | null = null
+      if (headId) {
+        head = await this.repositoriesStore.findGitHubRepositoryByID(headId)
+      }
+
+      // We know the base repo ID can't be null since it's the repository we
+      // fetched the PR from in the first place.
+      const baseId = pr.base.repoId!
+      const base = (await this.repositoriesStore.findGitHubRepositoryByID(
+        baseId
+      ))!
+
+      const prStatus = await this.getPullRequestStatusById(pr.head.sha, pr.id!)
```

Comment:
We're doing `pr.id!` in a couple of places here - can we centralize this and do some checks that it's valid? What's the risk if it's not a number?